### ISSUE TITLE
Warn when tsconfig.json extends a file that contains `references`

### ISSUE THREAD

**Issue by Woodz** [microsoft/TypeScript#61223]
### 🔎 Search Terms

tsconfig, extends, references

### 🕗 Version & Regression Information

This is intended behaviour according to https://www.typescriptlang.org/tsconfig/#extends

### 💻 Code

```ts
// Your code here
```

### 🙁 Actual behavior

I spent 2 days trying to get tsconfig.json file inheritance working with `references`. Then I found in the manual:

> Currently, the only top-level property that is excluded from inheritance is [references](https://www.typescriptlang.org/tsconfig/#references).

There was no warning in the command line when I was trying to extend a tsconfig.json file that contained `references` - instead it was just silently ignored.

### 🙂 Expected behavior

Should log a warning when extending a tsconfig.json file that contains `references`. This warning should explain that `references` are not inherited so therefore will be ignored

**Comment by RyanCavanaugh** [comment]
We generally don't do warnings like this unless the config is *definitely* wrong. It's entirely possible to imagine configurations where you're doing on this on purpose.
